### PR TITLE
Expose security filter to web-resources, prime permissions from state

### PR DESCRIPTION
**Pull request by jdillon**
This _should_ help prevent race to configure user and permissions when loading a feature page directly.

May need further optimization or consideration for the fetchpermissions command, but shouldn't hurt anything (sans some additional overhead) for the time being.

### CODE REVIEW

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Permissions.js [hunk 1]
```diff
@@ -59,12 +60,30 @@ Ext.define('NX.controller.Permissions', {
...
+    var me = this,
+        rawData = NX.State.getValue('permissions');
+
+    //<if debug>
+    me.logDebug('Initial permissions: ', rawData);
+    //</if>
 
-    me.fetchPermissions();
+    me.getStore('Permission').loadRawData(rawData, false);
+    NX.Permissions.setPermissions(me.getPermissions());
```
Comment (adreghiciu) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Permissions.js]: I do not think setting of permissions is necessary as #loadRawData will trigger a load event that will set the permissions and fire a permission change event

**Comment by adreghiciu** [comment]
+1

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Permissions.js (hunk 1 above)
Comment (jdillon) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Permissions.js]: I can test again, I wasn't seeing the event get fired, or it was lost in the async logging going on at the time, i'm not sure.

**Comment by kellyrob99** [comment]
+1

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF479-1 :white_check_mark:

**Comment by mcculls** [comment]
+1 looks reasonable

File: plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Permissions.js (hunk 1 above)
Comment (jdillon) [inline plugins/rapture/nexus-rapture-plugin/src/main/resources/static/rapture/NX/controller/Permissions.js]: @adreghiciu docs say:

    As of 4.2, this method will no longer fire the load event.

http://docs.sencha.com/extjs/4.2.3/#!/api/Ext.data.Store-method-loadRawData

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**